### PR TITLE
Remove all deprecated actions + Related code

### PR DESCRIPTION
**Pull request by mokagio**
## What does it do?

They became obsolete with the introduction of the `Versioning` module in d050d84 (#512) and as discussed in https://github.com/wordpress-mobile/release-toolkit/pull/579#pullrequestreview-2221043532 we ought to move forward in deleting them and ship a new major version.

Internal ref paaHJt-6gp-p2

## Checklist before requesting a review

- [x] Run `bundle exec rubocop` to test for code style violations and recommendations
- [x] Add Unit Tests (aka `specs/*_spec.rb`) if applicable
- [x] Run `bundle exec rspec` to run the whole test suite and ensure all your tests pass
- [x] Make sure you added an entry in [the `CHANGELOG.md` file](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/CHANGELOG.md#trunk) to describe your changes under the appropriate existing `###` subsection of the existing `## Trunk` section.
- [x] If applicable, add an entry in [the `MIGRATION.md` file](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/MIGRATION.md) to describe how the changes will affect the migration from the previous major version and what the clients will need to change and consider.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AliSoftware** [comment]
Overall LGTM, but I'd love to have @spencertransier chime in and give the final PR review/approval especially wrt the removal of the `lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_version_helper.rb` file, to confirm it's really not used anywhere in the `release-toolkit` codebase itself (e.g. by the `Versioning` module calling into it—I doubt it but better safe and double-check than sorry) or if it's used by any of our client repos, making sure that all the methods in here have a counterpart in the `Versioning` module they can be replaced with (I think they do, but again, better safe than sorry).

**Comment by mokagio** [comment]
I've been using this branch in https://github.com/Automattic/simplenote-android/pull/1676 and so far so good 😄

**Comment by spencertransier** [comment]
@mokagio Sorry I missed this! I will review it this week.

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_validate_ci_build.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ module Fastlane
   module Actions
     class IosValidateCiBuildAction < Action
       def self.run(params)
-        require_relative '../../helper/ios/ios_git_helper'
+        return_value '../../helper/git_helper'
```
Comment (spencertransier) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_validate_ci_build.rb]: Was this intended to be `require_relative` instead of `return_value`?

**Review by spencertransier — APPROVED** [review]
I went through and verified that all the removed methods are not used anywhere but in Simplenote Android/iOS/Mac, which I know you're already working on updating. I just [had a question](https://github.com/wordpress-mobile/release-toolkit/pull/580#discussion_r1739616310) about a `require_relative` vs `return_value`. Besides that, everything looks great. Approving to unblock.

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_validate_ci_build.rb (hunk 1 above)
Comment (spencertransier) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_validate_ci_build.rb]:
```suggestion
        require_relative '../../helper/git_helper'
```

Comment (mokagio) [reply lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_validate_ci_build.rb]: Wow... not sure how this ended up like that. Thanks @spencertransier !

Comment (mokagio) [reply lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_validate_ci_build.rb]: And, even better, thanks for doing the work to check this was unnecessary and deleting it via https://github.com/wordpress-mobile/release-toolkit/pull/586 !

**Pull request merged**